### PR TITLE
gpcheckcat leakedschema refactoring

### PR DESCRIPTION
**Pull request by kaknikhil**
Refactor leaked schema query into it's own module with its own set of unit tests . 

1. create a class for leaked schema handler 
2. refactor leaked schema unit tests 
3. address @danielgustafsson comments from PR #595

### CODE REVIEW

File: gpMgmt/Makefile [hunk 1]
```diff
@@ -82,7 +82,7 @@ install: generate_greenplum_path_file
 	mkdir -p $(prefix)/lib/python	
 	mkdir -p $(prefix)/sbin
 
-	#symlink gpcheckcat from bin to bin/lib to mainitain backward compatibility
+	#symlink gpcheckcat from bin to bin/lib to maintain backward compatibility
```
Comment (Chibin) [inline gpMgmt/Makefile]: This will be taken care of #634, so we can ignore this?

**Comment by Chibin** [comment]
LGTM

**Comment by jmcatamney** [comment]
Looks good to me.

File: gpMgmt/bin/gppylib/test/unit/test_unit_gpcheckcat.py [hunk 2]
```diff
@@ -83,16 +86,33 @@ def test_checkcat_report__after_running_unique_index_violations_check__reports_v
...
 
-    def test_drop_leaked_schemas__drops_orphaned_and_leaked_schemas(self):
-        self.db_connection.mock_add_spec(['close', 'query'])
-        self.subject.getLeakedSchemas = Mock(return_value=["fake_leak_1", "fake_leak_2"])
+    def test_drop_leaked_schemas__when_no_leaked_schemas_exist__passes_gpcheckcat(self):
+        self.subject.drop_leaked_schemas(self.leaked_schema_dropper, self.db_connection)
+
+        self.subject.setError.assert_not_called()
+
+    def test_drop_leaked_schemas____when_leaked_schemas_exist__finds_and_drops_leaked_schemas(self):
```
Comment (tangp3) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gpcheckcat.py]: Minor, __when / _when ?

**Comment by royc1** [comment]
Looks good to me.

**Comment by danielgustafsson** [comment]
Closing as this was pushed in 37e045f and 9dfaf11.

**Pull request closed without merging**